auth.php

```diff
@@ -104,6 +104,37 @@ public function __construct() {
         $this->defaultidp = auth_saml2_get_default_idp();
     }
 
+    public static function saml2_authproc_filters_hook() {
+        $authprocfilters = [];
+        $authprocfilters[50] = array(
+            'class' => 'core:AttributeMap',
+            'oid2name',
+        );
+        $callbacks = get_plugins_with_function('extend_auth_saml2_proc', 'lib.php');
+        foreach ($callbacks as $plugins) {
+            foreach ($plugins as $pluginfunction) {
+                $filters = $pluginfunction();
+                foreach ($filters as $key => $value) {
+                    $key = self::check_filters_priority($key, $authprocfilters);
+                    $authprocfilters[$key] = $value;
```

Comment:
can you please emit a debugger warning when two proc filters have the same key and clash